File: frontend/src/types/order.ts
export interface OrderItem {
  id: number;
  productId: number;
  quantity: number;
  price: number;
  lineTotal: number;
  productName: string;
  productCategory: string;
}

export interface Order {
  id: number;
  orderDate: string;
  totalAmount: number;
  status: string;
  paymentMethod: string;
  paymentStatus: string;
  totalItems: number;
}

export interface OrderDetail {
  id: number;
  userId: number;
  orderDate: string;
  totalAmount: number;
  status: string;
  shippingAddress: string;
  paymentMethod: string;
  paymentStatus: string;
  createdAt: string;
  items: OrderItem[];
}

export interface PlaceOrderRequest {
  shippingAddress: string;
  paymentMethod: string;
}

File: SupermarketAPI/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace SupermarketAPI.Models;

public enum OrderStatus
{
    Pending,
    Confirmed,
    Shipped,
    Delivered,
    Cancelled
}

public enum PaymentStatus
{
    Pending,
    Paid,
    Failed
}

public class Order
{
    public int Id { get; set; }

    [Required]
    public int UserId { get; set; }

    public DateTime OrderDate { get; set; } = DateTime.UtcNow;

    [Required]
    [Range(typeof(decimal), "0.00", "79228162514264337593543950335")]
    public decimal TotalAmount { get; set; }

    [Required]
    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    [Required]
    [StringLength(500)]
    public string ShippingAddress { get; set; } = string.Empty;

    [Required]
    [StringLength(50)]
    public string PaymentMethod { get; set; } = string.Empty;

    [Required]
    public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}